### ISSUE TITLE
[ISSUE] Importing Genesis beam fails in harmonic conversion 

### ISSUE THREAD

**Issue by eugenioferrari** [svenreiche/Genesis-1.3-Version4#155]
Hello,
Similar to what was happening with the field file, when doing an HGHG run and harmonic conversion without resampling, something similar happens when trying to import a particle file.

This is the error reported:
```*** Error: Mismatch in sample rate of run and of input file```

While I experience this with single slice simulations, I suspect the same issue is present in multislices.

**Comment by ZeugAusHH** [comment]
Dear Eugenio,

which version of GENESIS are you running? If possible please provide the version number -- or even better, the git commit id of the source code used to build the binary.

Best
Christoph

**Comment by eugenioferrari** [comment]
Dear Christoph,
I'm on the commit #9f9f00f.

Best
Eugenio

**Comment by ZeugAusHH** [comment]
Dear Eugenio,

thanks for posting the commit id, I was asking since I recently improved the error handling code in imports of fields/beams.
But since you're on the most recent commit, I think @svenreiche needs to take over.

Best
Christoph

**Comment by eugenioferrari** [comment]
Thank you for looking into this.
Attached you can find a minimal working example of the issue.

[example_issue.zip](https://github.com/svenreiche/Genesis-1.3-Version4/files/14113006/example_issue.zip)

And here is the error message:
```
Opened input file input_rad.in
Parsing lattice file lattice.lat ...
Setting up time window of 0.004 microns with 1 sample points...
Generating input radiation field for HARM = 1 ...
Importing particle distribution from file: modulated.par.h5 ...
*** Error: Mismatch between reference length of run and of input file

Program is terminating...
```

**Comment by eugenioferrari** [comment]
As suggested by @svenreiche, the issue is due to the discrepancy in the sampling rate. 
It can be solved by invoking an harmonic conversion via alter_setup before the actual tracking.

Thanks again!

**Comment by svenreiche** [comment]
Actually it is not that trivial. I tried to do it with alter_setup but then I am getting an error message. This time with the sample rate, since the dump has a sample rate of 75.
SO there is still some need to fix it.

**Comment by eugenioferrari** [comment]
Works for me when resample=false

**Comment by svenreiche** [comment]
Yes, I made it also work with setting explicit the sample to 75 in the second file for the time-window (resample should be per default false).  
The problem is more for steady-state simulation, since the dumped beam file still has a difference between reference length and slice length by 75. And that is not easy to define in the second run if it is also steady-state.  
So that needs to be fixed and I keep the issue open still in steady-state the sample rate is not changed.

**Comment by eugenioferrari** [comment]
Another point: it's also not working if the harmonic conversion is done in the modulator file, before the particle dump.

**Comment by ustc-cheese** [comment]
Hi Sven, Recently I have been running with the importbeam element but encountered the same error, which says, '*** Error: Mismatch between reference length of run and of input file.' I am not using any harmonic conversion but only a typical modulation. I have no idea of the reference length of the run and input file is. Could you please give an explanation?

**Comment by svenreiche** [comment]
Hi, 
In Genesis the reference value is the intrinsic unit of the longitudinal sampling (beside the sample rate). The particle coordinate are expressed in a ponderemotive phase, which scales the longitudinal position as theta = 2 pi s / reflength.
this value is defined in the setup namelist of each Genesis input deck.  If one run genesis and dumps it then this value is written as the reference length. If now this dump is used in a second run it cross checks if the reference length, defined in its own &setup namelist and the value in the dump file, which should match. Otherwise one would some wrong harmonic. conversion or frequency shift!!  Therefore this check is a sanity check.  

In a case of simulating HGHG this might occur in the conversion of Modulator + Chicane -> Radiator.   Somewhere one has to do a harmonic conversion.   The problem (and the reason of this issue is that the behavior is differently for steady-state and time-dependent running.

If I have time I should look into this issue. If you still thinks there is a bug, please send me your input decks (attach to this thread) so that I can cross check and fix it (if it is a bug).

Best
Sven

**Comment by ustc-cheese** [comment]
Hi Sven,
Here are my input and beamfile
&setup
rootname=adm
lattice=adm.lat
beamline=mod
lambda0=266e-09
gamma0=2739.726
delz=0.01
shotnoise=1
one4one=false
npart=32000
nbins=64
&end

&time
sample=100
time=true
&end

&importbeam
file=beam.h5
time=true
&end

&profile_gauss
label=prof1
c0=600e6
s0=50e-6
sig=15e-6
&end

&field
power=@prof1
dgrid=2.000000e-3
ngrid=301
waist_size=0.00133227
waist_pos=0.8
&end

&track
&end
with a lat
M1: Marker = { dumpbeam=1 };
M2: Marker = { dumpbeam=1 };


LSRM: UNDULATOR = { lambdau = 0.06, nwig = 20, aw = 2.64175 };

WIG2: UNDULATOR = { lambdau = 0.02, nwig = 50, aw = 2.695491 };


mod: LINE = { LSRM, M1 };

line2: LINE = { WIG2, M2 };
and an elegant2genesis file converted to beam.h5 with 200 slices
Best 
Weijie

**Comment by svenreiche** [comment]
HI Weijie,

If you want to import from Elegant by using the slice parameters from elegant2genesis then the best approach would be to use the namelists profile_file or profile_file_multi and then refer to the profiles in the beam namelist. Importbeam requires the internal format of Genesis and works only if you have made a dump in a previous run with Genesis. 

You might need to update Genesis since I fixed profile_file_multi about 6 months ago.

Best
Sven
